lib/llvmopencl/Barrier.h

```diff
@@ -85,6 +86,23 @@ namespace pocl {
         (llvm::CallInst::Create(F, "", InsertBefore));
     }
 
+#else
+    static Barrier *create(InstListType::iterator InsertBefore) {
```

Comment:
Do you need to duplicate the whole function? Could typedef work here?

Reply:
minimized the diff